src/golang/lib/dynamic/engine.go

```diff
@@ -318,5 +338,5 @@ func ResyncClusterState(
 	// inconsistent state between the database and terraform. In this case, we resync the state by
 	// deleting the cluster and updating the database state to be Terminated.
 	log.Error("Dynamic k8s cluster might be in an inconsistent state. Resolving state by deleting the cluster...")
-	return DeleteDynamicEngine(ctx, engineIntegration, integrationRepo, db)
+	return DeleteDynamicEngine(ctx, true, engineIntegration, integrationRepo, db)
```

Comment:
nit: Can you add an inline comment `true, // skipPodsStatusCheck` for clarity? Each arg would go on it's own line then.